### DISCUSSION TITLE
$(GitBranch) prop value is empty in .csproj file

### DISCUSSION THREAD

**Discussion by sanjaykulkarni04** [devlooped/GitInfo#177] [Q&A]
Hello,

While building csproj locally/CI server on **_master_** branch, I wanted to add RC NuGet package reference otherwise PRE release NuGet package reference. Please find the below code added in .csproj  file. What's wrong with the below code? Can someone assist me?

```
<Project Sdk="Microsoft.NET.Sdk">
  <PropertyGroup>
    <TargetFramework>net5.0</TargetFramework>
    <GitBranch>$(GitBranch.Trim())</GitBranch>
  </PropertyGroup>

  <ItemGroup>
	  <PackageReference Include="AutoMapper" Version="10.1.1" />
	  <PackageReference Include="GitInfo" Version="2.2.0" />
    <PackageReference Include="Data.Foundation.Messaging" Version="4.1.5" />    
  </ItemGroup>

	<Choose>
		<When Condition="$(GitBranch) == 'master'">
			<ItemGroup>
				<PackageReference Include="Data.Account.Domain.Messaging" Version="1.0.0-rc*" IncludePrerelease="true" />
			</ItemGroup>
		</When>
		<Otherwise>
			<ItemGroup>
				<PackageReference Include="Data.Account.Domain.Messaging" Version="1.0.0-pre*" IncludePrerelease="true" />
			</ItemGroup>
		</Otherwise>
	</Choose>
  <ItemGroup>
    <ProjectReference Include="..\Data.Account.GraphQl.Schema\Data.Account.GraphQl.Schema.csproj" />
  </ItemGroup>
  
</Project>
```

**Comment by Farnam88** [comment]
> Hello,
> 
> While building csproj locally/CI server on **_master_** branch, I wanted to add RC NuGet package reference otherwise PRE release NuGet package reference. Please find the below code added in .csproj file. What's wrong with the below code? Can someone assist me?
> 
> ```
> <Project Sdk="Microsoft.NET.Sdk">
>   <PropertyGroup>
>     <TargetFramework>net5.0</TargetFramework>
>     <GitBranch>$(GitBranch.Trim())</GitBranch>
>   </PropertyGroup>
> 
>   <ItemGroup>
> 	  <PackageReference Include="AutoMapper" Version="10.1.1" />
> 	  <PackageReference Include="GitInfo" Version="2.2.0" />
>     <PackageReference Include="Data.Foundation.Messaging" Version="4.1.5" />    
>   </ItemGroup>
> 
> 	<Choose>
> 		<When Condition="$(GitBranch) == 'master'">
> 			<ItemGroup>
> 				<PackageReference Include="Data.Account.Domain.Messaging" Version="1.0.0-rc*" IncludePrerelease="true" />
> 			</ItemGroup>
> 		</When>
> 		<Otherwise>
> 			<ItemGroup>
> 				<PackageReference Include="Data.Account.Domain.Messaging" Version="1.0.0-pre*" IncludePrerelease="true" />
> 			</ItemGroup>
> 		</Otherwise>
> 	</Choose>
>   <ItemGroup>
>     <ProjectReference Include="..\Data.Account.GraphQl.Schema\Data.Account.GraphQl.Schema.csproj" />
>   </ItemGroup>
>   
> </Project>
> ```

I have the same problem!

**Reply by kzu** [reply]
You're trying to accesss/condition on a property whose value is calculated in a target. That's never going to work. You cannot add conditional package references via targets either, so this scenario cannot be implemented as outlined, I'm afraid. You will need to pre-calculate the branch beforehand (i.e. pass it via envvars or cli).
